tests/unit/test_agw_preinstall_checks.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import unittest
+from unittest.mock import PropertyMock, call, patch
+
+from magma_access_gateway_installer.agw_installation_errors import (
+    InvalidNumberOfInterfacesError,
+    UnsupportedOSError,
+)
+from magma_access_gateway_installer.agw_preinstall_checks import (
+    AGWInstallerPreinstallChecks,
+)
+
+
+class TestAGWInstallerPreinstallChecks(unittest.TestCase):
+
+    TEST_NETWORK_INTERFACES = ["eth0", "eth1"]
+
+    def setUp(self) -> None:
+        self.agw_preinstall_checks = AGWInstallerPreinstallChecks(self.TEST_NETWORK_INTERFACES)
+
+    @patch(
+        "magma_access_gateway_installer.agw_preinstall_checks.AGWInstallerPreinstallChecks._ubuntu_is_installed",  # noqa: E501
+        new_callable=PropertyMock,
+    )
+    @patch(
+        "magma_access_gateway_installer.agw_preinstall_checks.AGWInstallerPreinstallChecks._required_amount_of_network_interfaces_is_available",  # noqa: E501
+        new_callable=PropertyMock,
+    )
+    def test_given_preinstall_checks_when_os_is_not_ubuntu_then_unsupported_os_error_is_raised(
```

Comment:
```suggestion
    def test_given_os_is_not_ubuntu_when_preinstall_checks_then_unsupported_os_error_is_raised(
```

Reply:
This same mixup of given/when is in other test names as well

Reply:
Fixed.